packages/auth/src/platform_browser/strategies/phone.ts

```diff
@@ -65,14 +78,92 @@ class ConfirmationResultImpl implements ConfirmationResult {
     );
     return this.onConfirmation(authCredential);
   }
+
+  async confirmWithWebOTP(
+    auth: Auth,
+    webOTPTimeout: number
+  ): Promise<UserCredential | undefined> {
+    if ('OTPCredential' in window) {
+      const abortController = new AbortController();
+      const timer = setTimeout(() => {
+        abortController.abort();
+
+        const myErr = _errorWithCustomMessage(
+          auth,
+          AuthErrorCode.WEB_OTP_NOT_RETRIEVED,
+          `Web OTP code is not fetched before timeout`
+        ) as WebOTPError;
+        myErr.confirmationResult = this;
+        throw myErr;
+      }, webOTPTimeout * 1000);
+
+      // @ts-ignore - ignore types for testing
+      const o: OTPCredentialRequestOptions = {
+        otp: { transport: ['sms'] },
+        signal: abortController.signal
+      };
+
+      let code: string = '';
+      await (
+        window.navigator['credentials'].get(o) as Promise<OTPCredential | null>
+      )
+        .then(async content => {
+          if (
+            content === undefined ||
+            content === null ||
+            content.code === undefined
+          ) {
+            const myErr = _errorWithCustomMessage(
+              auth,
+              AuthErrorCode.WEB_OTP_NOT_RETRIEVED,
+              `Web OTP code is not valid`
```

Comment:
Should be a different error message than "code is not valid"?

Reply:
Would "the atuo-retrieved credential or code is not defined" be a better error message?

Reply:
Sounds good to me.